ddraw/IDirect3DDeviceX.cpp

```diff
@@ -2453,11 +2542,98 @@ HRESULT m_IDirect3DDeviceX::DrawIndexedPrimitive(D3DPRIMITIVETYPE dptPrimitiveTy
 		}
 		else
 		{
+			const UINT stride = GetVertexStride(dwVertexTypeDesc);
+
+			// Handle PositionT
+			if((dwVertexTypeDesc & D3DFVF_XYZRHW) != 0 && Config.DdrawConvertHomogeneousW)
```

Comment:
Instead of this, check `(dwVertexTypeDesc & 0x0E) == D3DFVF_XYZRHW` to avoid mix-ups with `D3DFVF_XYZB1` and up.

Reply:
Good catch.  Fixed in the latest check in.